app/controllers/api/jobs_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Api::JobsController < ApplicationController
+  def create
+    job = Job.where(job_params).first_or_initialize
```

Comment:
Maybe not including all job params would work better? (in particular I'm think if we leave off "source" we wouldn't have duplicates from multiple job boards).